crates/driver/src/infra/liquidity/config.rs

```diff
@@ -93,6 +93,20 @@ impl UniswapV2 {
             missing_pool_cache_time: Duration::from_secs(60 * 60),
         })
     }
+
+    /// Returns the liquidity configuration for liquidity sources only used on
+    /// test networks.
+    pub fn testnet_uniswapv2(network: &eth::NetworkId) -> Option<Self> {
+        Some(Self {
+            router: deployment_address(
+                contracts::TestnetUniswapV2Router02::raw_contract(),
```

Comment:
How is this router different from the standard uni v2 router? Could we theoretically configure it on the defaults uniswap preset conditionally on Sepolia's network?

Reply:
Router, factory and pair are exactly the same as on vanilla mainnet except for two things:
- The Sepolia factory has an extra line `bytes32 public constant INIT_CODE_HASH = keccak256(abi.encodePacked(type(UniswapV2Pair).creationCode));`.
- the [compilation metadata](https://docs.soliditylang.org/en/v0.8.23/metadata.html#encoding-of-the-metadata-hash-in-the-bytecode) is different, which in turn causes the init hash to be different, but this has no impact on the actual functionality of the contract.

> Could we theoretically configure it on the defaults uniswap preset conditionally on Sepolia's network?

No because the init hash is different. and that's used to determine the address of a pair given the tokens in a pair. It's also why I removed the Uniswap deployment on Gnosis Chain.